### PR TITLE
ValidatingVisibility Event

### PR DESCRIPTION
**Pull request by Misfiy**

### CODE REVIEW

**Review by Misaka-ZeroTwo — COMMENTED** [review]
Fix build errors

File: Exiled.Events/EventArgs/Scp939/ValidatingVisibilityEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.EventArgs.Scp939
+{
+    using API.Features;
+    using Exiled.API.Features.Roles;
+    using Interfaces;
+
+    /// <summary>
+    ///     Contains all information before SCP-939 plays a stolen player's voice.
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Scp939/ValidatingVisibilityEventArgs.cs:15]: Copy pasta docs moment

File: Exiled.Events/EventArgs/Scp939/ValidatingVisibilityEventArgs.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    /// </summary>
+    public class ValidatingVisibilityEventArgs : IScp939Event, IDeniableEvent
```
Comment (Misaka-ZeroTwo) [inline Exiled.Events/EventArgs/Scp939/ValidatingVisibilityEventArgs.cs:17]: Validating or validated?

**Review by Misaka-ZeroTwo — CHANGES_REQUESTED** [review]

**Review by louis1706 — APPROVED** [review]

File: Exiled.Events/EventArgs/Scp939/ValidatingVisibilityEventArgs.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.EventArgs.Scp939
+{
+    using API.Features;
+    using Exiled.API.Features.Roles;
+    using Interfaces;
+
+    /// <summary>
+    ///     Contains all information about players before being validated to SCP-939.
```
[the remark this answers is not in the document]
Comment (Misaka-ZeroTwo) [reply Exiled.Events/EventArgs/Scp939/ValidatingVisibilityEventArgs.cs:15]: Validating doesn't sound intuitive to me.

File: Exiled.Events/Handlers/Scp939.cs [hunk 4]
```diff
@@ -94,5 +99,11 @@ public static class Scp939
         /// </summary>
         /// <param name="ev">The <see cref="ClawedEventArgs"/> instance.</param>
         public static void OnClawed(ClawedEventArgs ev) => Clawed.InvokeSafely(ev);
+
+        /// <summary>
+        /// Called when a players visibility is being validated.
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp939.cs:104]:
```suggestion
        /// Called before validating visibility.
```

File: Exiled.Events/Handlers/Scp939.cs [hunk 5]
```diff
@@ -53,6 +53,11 @@ public static class Scp939
         /// <remarks>This event is calling only when attack doesn't have target.</remarks>
         public static Event<ClawedEventArgs> Clawed { get; set; } = new();
 
+        /// <summary>
+        /// Invoked when checking visibility for players.
```
Comment (NaoUnderscore) [inline Exiled.Events/Handlers/Scp939.cs:57]:
```suggestion
        /// Invoked before validating visibility.
```

File: Exiled.Events/EventArgs/Scp939/ValidatingVisibilityEventArgs.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        /// <summary>
+        ///     Gets the player who's controlling SCP-939.
+        /// </summary>
+        public Player Player { get; }
+
+        /// <inheritdoc/>
+        public Scp939Role Scp939 { get; }
+
+        /// <summary>
+        ///    Gets or sets a value indicating whether or not SCP-939 is allowed to view the player.
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/Scp939/ValidatingVisibilityEventArgs.cs:53]:
```suggestion
        ///    Gets or sets a value indicating whether visibility can be validated.
```

File: Exiled.Events/EventArgs/Scp939/ValidatingVisibilityEventArgs.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        public ValidatingVisibilityEventArgs(ReferenceHub player, ReferenceHub target, bool isAllowed)
+        {
+            Player = Player.Get(player);
+            Scp939 = Player.Role.As<Scp939Role>();
+            Target = Player.Get(target);
+            IsAllowed = isAllowed;
+        }
+
+        /// <summary>
+        ///     Gets the player being shown to SCP-939.
```
Comment (NaoUnderscore) [inline Exiled.Events/EventArgs/Scp939/ValidatingVisibilityEventArgs.cs:40]:
```suggestion
        ///     Gets the player who's being shown to SCP-939.
```

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

**Pull request merged**